src/components/client/Layout.js

```diff
@@ -1,48 +1,163 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import SideNav from './containers/SideNav';
 import M from 'materialize-css';
+import { connect } from 'react-redux';
+import { getSearchProducts } from '../../helpers/api';
+import { saveSearchResult } from '../../actions/example.actions';
+import './Layout.css';
+import logo from './../../images/logo_storeo_white.png';
+
 
 class Layout extends Component {
+  state = {
+    selectedCategoryName: undefined,
+    categoryId: null,
+    queryPhrase: '',
+  }
 
   componentDidMount() {
     M.AutoInit();
+    // this.totalBasketQty();
+  }
+
+  totalBasketQty = () => {
+    let subTotalBasketQty = 0;
+    this.props.basket.forEach(item => {
+      subTotalBasketQty += item.quantity
+    });
+    return subTotalBasketQty;
+  }
+
+  selectCategory = async (categoryId, selectedCategoryName) => {
+    await this.setState({ categoryId, selectedCategoryName });
+  }
+
+  handleSearchSubmit = async (event) => {
+    event.preventDefault();
+    const { queryPhrase, selectedCategoryName } = this.state;
+    let categoryQuery;
+    if (!selectedCategoryName) categoryQuery = '';
+    else categoryQuery = '&category=' + selectedCategoryName;
+
+    const searchQuery = `?q=${queryPhrase}${categoryQuery}`;
+    let searchResult = await getSearchProducts(searchQuery);
+    this.props.saveSearchResult(searchResult.data);
+
+  }
+
+  handleChange = (event) => {
+    event.preventDefault();
+    this.setState({ queryPhrase: event.target.value });
   }
 
   render() {
+
     return (
+
       <React.Fragment>
```

Comment:
This template is massive. One of the things JSX is helping us with is to give semantics to the HTML-like structure. Creating thin components can improve readability. This is why Styled Components or Emotion are good options, they let us create this thin components avoiding some boilerplate.